src/azul/service/responseobjects/storage_service.py

```diff
@@ -37,14 +43,144 @@ def put(self, object_key: str, data: bytes, content_type: Optional[str] = None)
 
         return object_key
 
+    @contextmanager
+    def multipart_upload(self, object_key: str, content_type: str):
+        api_response = self.client.create_multipart_upload(Bucket=self.__bucket_name,
+                                                           Key=object_key,
+                                                           ContentType=content_type)
+        upload_id = api_response['UploadId']
+
+        with ThreadPoolExecutor(max_workers=MULTIPART_UPLOAD_MAX_WORKERS) as thread_pool:
+            handler = MultipartUploadHandler(self.__bucket_name, object_key, upload_id, thread_pool)
+
+            try:
+                yield handler
+            except Exception as e:
+                logger.error('Upload %s: Aborting due to unexpected error', upload_id)
+                handler.abort()
+                logger.warning('Upload %s: Aborted', upload_id)
+                raise UnexpectedMultipartUploadAbort(f'{type(e).__name__}: {e}')
+
+            if handler.is_active:
+                handler.complete()
+
     def get_presigned_url(self, key: str) -> str:
         return self.client.generate_presigned_url(ClientMethod='get_object',
                                                   Params=dict(Bucket=self.__bucket_name, Key=key))
 
     def create_bucket(self, bucket_name: str = None):
         self.client.create_bucket(Bucket=(bucket_name or self.__bucket_name))
-        logger.warning(f'{type(self).__name__}: Created a bucket called "{bucket_name or self.__bucket_name}"')
+
+
+class MultipartUploadHandler:
+
+    def __init__(self, bucket_name, object_key, upload_id, thread_pool):
+        self.__resource = boto3.resource('s3')
+        self.__bucket_name = bucket_name
+        self.__object_key = object_key
+        self.__upload_id = upload_id
+        self.__handler = self.__resource.MultipartUpload(self.__bucket_name, self.__object_key, self.__upload_id)
```

Comment:
```suggestion
        self.mp_upload = self.__resource.MultipartUpload(self.__bucket_name, self.__object_key, self.__upload_id)
```

It's really confusing that a class whose name ends in `Handler` has a field called `handler`. In general you should avoid calling things just `handler` because that's just too generic.